src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -9514,6 +9514,360 @@ .maxstack  6
   IL_0032:  call       ""void S.M(S, CustomHandler)""
   IL_0037:  ret
 }
+");
+
+            comp = CreateCompilation(new[] { code, InterpolatedStringHandlerArgumentAttribute, handler }, options: TestOptions.DebugExe);
+            CompileAndVerify(comp, expectedOutput: @"
+s1.I:1
+s2.I:2");
+        }
+
+        [Fact, WorkItem(58514, "https://github.com/dotnet/roslyn/issues/58514")]
+        public void StructReceiver_Rvalue_ObjectCreationReceiver_01()
```

Comment:
>StructReceiver_Rvalue_ObjectCreationReceiver_01 [](http://example.com/codeflow?start=20&length=47)

Are we testing scenario with the other sequence we create in the ```addInterpolationPlaceholderReplacements``` helper? I think it is created for an argument rather than for a receiver. It would be good to cover different ref-ness scenarios for the argument. Also, I think it would be good to make sure we have a test for a scenario where the receiver is a reference type rather than a struct.

Reply:
> Are we testing scenario with the other sequence we create in the `addInterpolationPlaceholderReplacements` helper?

Yes. This one never had a problem because of the way the previous argument spilling worked, but it's tested in `StructArgument_Rvalue_ObjectCreationArgument`. I can add the other tests.